src/first_app.md

```diff
@@ -26,6 +33,16 @@ In Relm4, a message can be any data type, but most often, an `enum` is used. In
 {{#include ../examples/simple_manual.rs:msg }}
 ```
 
+Computers are capable of both sending and receiving messages, and so, components in Relm4 can send and receive messages to and from themselves and other components.
+
+This is accomplished by having two types of messages:
+- **Input** messages are sent and received by components and handled in the `update` function.
+- **Output** messages are sent by components to other components. 
+
+You can send output messages using the `output_to_parent_msg` function from the sender component and receive them in the `forward` function inside the receiver component.
```

Comment:
You're absolutely right!

Reply:
I updated the doc to explain the difference a bit better, let me know your thoughts.

Reply:
I like the it in general, but I think the first app chapter should focus only on getting started with a simple app. I would at most mention components and factories and put all the explanations into their dedicated chapters.

Reply:
I'm working on abstracting the core components of Relm4 into different chapters to be read before even making an app.

![image](https://user-images.githubusercontent.com/22224438/168771358-54ccd316-fa8a-4c5b-8093-49b8b892e69b.png)

There we will explain everything to detail and by the time they reach the simple app, it will be easier to understand.

Reply:
Sounds good!